subsys/bootloader/CMakeLists.txt

```diff
@@ -171,19 +171,92 @@ include (${CMAKE_CURRENT_LIST_DIR}/cmake/debug_keys.cmake)
 
 set(sign_depends kernel_elf ${key_file_depends})
 
+set(hash_file firmware.sha256)
+set(signature_file firmware.signature)
+set(public_key_file public.pem)
+
+set(hashcmd
+  ${PYTHON_EXECUTABLE}
+  ${NRF_BOOTLOADER_SCRIPTS}/hash.py --in ${PROJECT_BINARY_DIR}/${KERNEL_BIN_NAME} > ${hash_file}
+  )
+
+if (${CONFIG_SB_SIGNING_PYTHON})
+  set(signcmd
+    ${PYTHON_EXECUTABLE}
+    ${NRF_BOOTLOADER_SCRIPTS}/do_sign.py --private-key ${signature_private_key_file} --in ${hash_file} > ${signature_file}
+    )
+  set(pubgencmd
+    ${PYTHON_EXECUTABLE}
+    ${NRF_BOOTLOADER_SCRIPTS}/keygen.py --public --in ${signature_private_key_file} --out ${public_key_file}
+    )
+elseif (${CONFIG_SB_SIGNING_OPENSSL})
```

Comment:
AFAIK you don't need `${...}` around these kconfig variables (or any other) when checking them.